client/src/app/components/lesson/video-container/video-container.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, ElementRef, Input, DoCheck, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { VideoService } from '../../../services/video.service';
+
+@Component({
+  selector: 'app-video-container',
+  templateUrl: './video-container.component.html',
+  styleUrls: ['./video-container.component.css']
+})
+export class VideoContainerComponent implements OnInit, DoCheck, OnDestroy {
+  @ViewChild('video', { static: true }) videoRef: ElementRef<HTMLVideoElement>;
+  @Input() public videoId: string;
+  @Input() public videoLink: string | undefined;
+  public currentSpeed?: number;
+  public isLoading = false;
+  public videoStreamError = false;
+
+  constructor(private videoService: VideoService) { }
+
+  public async ngOnInit(): Promise<void> {
+    // Initially pass videoRef into VideoService
+    this.videoService.videoRef = this.videoRef;
+    this.currentSpeed = this.videoRef.nativeElement.playbackRate;
+
+    // Don't allow to execute code below if pictureInPicture still in the document
+    if (document.pictureInPictureElement) {
+      return;
+    }
+
+    // Check if there is data in the LocalStorage with timeline for the video
+    const videoData = localStorage.getItem(this.videoId);
+    if (videoData) {
+      const { link, currentTime } = JSON.parse(videoData);
+      this.videoService.setVideoLink(link);
+      await this.videoService.runVideoStream(link, this.videoRef, +currentTime);
+      return;
+    }
```

Comment:
```suggestion
    const videoData = localStorage.getItem(this.videoId);
    
    if (videoData) {
      const { link, currentTime } = JSON.parse(videoData);
      
      this.videoService.setVideoLink(link);
      await this.videoService.runVideoStream(link, this.videoRef, +currentTime);
      
      return;
    }
```
It's better when you separate logic in your blocks
![image](https://user-images.githubusercontent.com/63243136/232597709-b3ff63e8-af5e-4171-af8e-f611950ad06f.png)